src/app/features/collectibles/components/ordinal-inscriptions.tsx

```diff
@@ -78,25 +79,37 @@ function Inscription({ path }: InscriptionProps) {
   }
 }
 
-interface InscriptionLoaderProps {
-  txid: string;
-  children(path: string): JSX.Element;
-}
-function InscriptionLoader({ txid, children }: InscriptionLoaderProps) {
-  const { data: inscriptionDetails } = useInscriptionByTxidQuery(txid);
-  if (!inscriptionDetails) return null;
-  return children(inscriptionDetails.inscriptions);
-}
-
-export function Ordinals() {
+export function OrdinalInscriptions() {
   const { data: utxos = [] } = useTaprootAddressUtxosQuery();
 
+  // use each utxo to get inscription details
+  const inscriptionInfoQueries = useQueries({
+    queries: utxos.map(utxo => createQueryOptionsInscriptionInfo(utxo.txid)),
+  });
+
+  // use inscription details to get inscription metadata
+  const inscriptionMetadataQueries = useQueries({
+    queries: inscriptionInfoQueries
+      .filter((q): q is DefinedQueryObserverResult<OrdApiXyzGetTransactionOutput> =>
+        Boolean(q.data)
+      )
+      .map(q => createQueryOptionsInscriptionMetadata(q.data.inscriptions)),
+  });
+
+  // use metadata to sort by descending inscription id
+  const sorted = inscriptionMetadataQueries
+    .filter((q): q is DefinedQueryObserverResult<OrdApiXyzGetInscriptionByInscriptionSchema> =>
+      Boolean(q.data)
+    )
+    .sort((q1, q2) => {
+      if (q1.data.inscription_number < q2.data.inscription_number) return 1;
+      return -1;
```

Comment:
Can't this be an expression like ?
```suggestion
      return q1.data.inscription_number < q2.data.inscription_number;
```

Reply:
Instead of a boolean, the `sort()` method is expected to [return a number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#description)

Reply:
ah yeah, then `q1.data.inscription_number - q2.data.inscription_number`?